apps/Steam/install-64

```diff
@@ -40,36 +40,36 @@ else
     sudo cp /tmp/sources.list /etc/apt/sources.list
     error "i386 architecture caused apt to error so it has been removed!"
   fi
+fi
+
+# check if any i386 packages are available from the current sources, if not, then add repos
+if apt-cache policy libc6:i386 --quiet=0 2>&1 | grep -q "N: Unable to locate package\|Candidate: (none)" ; then
+
+  # check if upstream-release is available
+  if [ -f /etc/upstream-release/lsb-release ]; then
+    # this is a derivative, get the upstream release info
+    distributor=$(lsb_release -sui)
+    codename=$(lsb_release -suc)
+  else
+    distributor=$(lsb_release -si)
+    codename=$(lsb_release -sc)
+  fi
 
-  # check if any i386 packages are available from the current sources, if not, then add repos
-  if apt-cache policy libc6:i386 --quiet=0 2>&1 | grep -q "N: Unable to locate package\|Candidate: (none)" ; then
-
-    # check if upstream-release is available
-    if [ -f /etc/upstream-release/lsb-release ]; then
-      # this is a derivative, get the upstream release info
-      distributor=$(lsb_release -sui)
-      codename=$(lsb_release -suc)
-    else
-      distributor=$(lsb_release -si)
-      codename=$(lsb_release -sc)
-    fi
-
-    if [[ "$distributor" == "Ubuntu" ]]; then
-      echo "deb [arch=amd64,i386] http://archive.ubuntu.com/ubuntu $codename main restricted universe multiverse
+  if [[ "$distributor" == "Ubuntu" ]]; then
+    echo "deb [arch=amd64,i386] http://archive.ubuntu.com/ubuntu $codename main restricted universe multiverse
 deb [arch=amd64,i386] http://archive.ubuntu.com/ubuntu $codename-updates main restricted universe multiverse
 deb [arch=amd64,i386] http://archive.ubuntu.com/ubuntu $codename-backports main restricted universe multiverse
 deb [arch=amd64,i386] http://archive.ubuntu.com/ubuntu $codename-security main restricted universe multiverse" | sudo tee /etc/apt/sources.list.d/steam-ubuntu-x86.list
-    else
-      error "You are using an unsupported debian based release and do not have i386 packages available. Steam installation will fail so this script will exit now."
-    fi
-
-    # perform an apt update to check for errors
-    # if apt update errors, assume that adding the additional sources.list caused it and remove it
-    sudo apt update
-    if [[ "$?" != 0 ]]; then
-      sudo rm -f /etc/apt/sources.list.d/steam-ubuntu-x86.list
-      error "/etc/apt/sources.list.d/steam-ubuntu-x86.list caused apt to error so it has been removed!"
-    fi
+  else
+    error "You are using an unsupported debian based release and do not have i386 packages available. Steam installation will fail so this script will exit now."
+  fi
+
+  # perform an apt update to check for errors
+  # if apt update errors, assume that adding the additional sources.list caused it and remove it
+  sudo apt update
+  if [[ "$?" != 0 ]]; then
+    sudo rm -f /etc/apt/sources.list.d/steam-ubuntu-x86.list
```

Comment:
@theofficialgman should'nt this also remove i386 multiarch with `dpkg --remove-architecture i386`

Reply:
no. not in my mind at least. i386 could or could not have already been enabled on the users install, and it is confirmed to not have caused apt to error (based on the previous sudo apt update check a few lines above)

Reply:
but i386 arch without proper sources is not good. so if removing i386 sources remove multiarch also

Reply:
fair assessment. changing